cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -319,7 +321,10 @@ func (c *Client) GetMemberID(peerURL string) (uint64, error) {
 
 // ListMembers returns the member list.
 func (c *Client) ListMembers() ([]Member, error) {
-	resp, err := c.listMembers()
+	if c.listMemberFunc == nil {
+		c.listMemberFunc = c.listMembers
+	}
+	resp, err := c.listMemberFunc()
```

Comment:
we should move this in the function
`func New(endpoints []string, ca, cert, key string) (*Client, error) {`